### ISSUE TITLE
Replacing the default boto3 session with a custom one

### ISSUE THREAD

**Issue by adhorn** [awslabs/aws-lambda-powertools-python#704]
**Is your feature request related to a problem? Please describe.**

I would like to replace the default boto3 powertools client and provide my own boto3 session to record local calls using [Placebo](https://github.com/garnaat/placebo)

**Describe the solution you'd like**

For example, here's how you could inject a Placebo client for local tests:
```
import placebo
import boto3
from aws_lambda_powertools.utilities import Parameters

session = boto3.Session()
pill = placebo.attach(session, data_path=PLACEBO_PATH)
pill.playback()
client = session.client('appconfig')
Parameters.set_boto3_client(client)
```

something like this...?

**Comment by heitorlessa** [comment]
This is now available as part of 1.21.0 release: https://github.com/awslabs/aws-lambda-powertools-python/releases/tag/v1.21.0

We've added others besides the parameters utility ;)

**Comment by adhorn** [comment]
Legends!
